docs/data/pages.ts

```diff
@@ -80,8 +80,8 @@ const pages: MuiPage[] = [
           '/x/react-data-grid/pivoting',
         ],
         children: [
-          { pathname: '/x/react-data-grid/row-grouping', plan: 'premium' },
           { pathname: '/x/react-data-grid/tree-data', plan: 'pro' },
+          { pathname: '/x/react-data-grid/row-grouping', plan: 'premium' },
           { pathname: '/x/react-data-grid/aggregation', title: 'Aggregation', plan: 'premium' },
           { pathname: '/x/react-data-grid/pivoting', title: 'Pivoting 🚧', plan: 'premium' },
```

Comment:
I have sorted the side nav with the first community, second pro, and third premium. The idea is that:

1. docs usually display the basic features first, so we can communicate that Premium is more advanced than Pro and Pro more advanced than Community.
2. It separates a bit more MIT from the Commercial features, giving more of the sense that MIT can be used standalone.